frontend/awx/resources/templates/WorkflowVisualizer/hooks/useWorkflowVisualizerToolbarActions.tsx

```diff
@@ -5,19 +5,69 @@ import {
   CheckCircleIcon,
   CloseIcon,
   CompressAltIcon,
+  EllipsisVIcon,
   ExpandAltIcon,
   PlusCircleIcon,
+  TrashIcon,
 } from '@patternfly/react-icons';
-import { Badge, Button, ToolbarGroup, ToolbarItem } from '@patternfly/react-core';
+import {
+  Badge,
+  Button,
+  Dropdown,
+  DropdownItem,
+  DropdownList,
+  MenuToggle,
+  ToolbarGroup,
+  ToolbarItem,
+} from '@patternfly/react-core';
 import { WorkflowNode } from '../../../../interfaces/WorkflowNode';
-import { usePageNavigate } from '../../../../../../framework';
+import { useBulkConfirmation, usePageNavigate } from '../../../../../../framework';
 import { AwxRoute } from '../../../../AwxRoutes';
+import { getItemKey, requestDelete } from '../../../../../common/crud/Data';
+import { stringIsUUID } from '../../../../common/util/strings';
 
 export function useWorkflowVisualizerToolbarActions(nodes: WorkflowNode[]) {
   const { t } = useTranslation();
   const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [isKebabOpen, setIsKebabOpen] = useState<boolean>(false);
   const pageNavigate = usePageNavigate();
   const params = useParams<{ id: string }>();
+  const bulkAction = useBulkConfirmation<WorkflowNode>();
+  const handleDeleteAllNodes = useCallback(
+    (workflowNodes: WorkflowNode[]) => {
+      return bulkAction({
+        title: t('Delete all workflow nodes'),
+        items: workflowNodes,
+        keyFn: getItemKey,
+        isDanger: true,
+        actionFn: (workflowNode: WorkflowNode) =>
```

Comment:
This function signature is ```  actionFn: (item: T, signal: AbortSignal) => Promise<unknown>;```
```suggestion
        actionFn: (workflowNode: WorkflowNode, signal: AbortSignal) =>
```

Reply:
The signal: AbortSignal, part of this type definition is supplied to the actionFn via requestDelete.  I do not have to define it here.

> This function signature is ` actionFn: (item: T, signal: AbortSignal) => Promise<unknown>;`

The signal: AbortSignal, part of this type definition is supplied to the actionFn via requestDelete.  I do not have to define it here.

Reply:
I believe we do not pass the abortSignal here because is not a required parameter. 

I talked with @jamestalton and we should make the signal required, work for a different PR.

```
export async function requestDelete<ResponseBody>(
  url: string,
  signal?: AbortSignal
): Promise<ResponseBody>

```

> signal should be required, otherwise devs will not actually hook it up
in addition, the bulk dialog creates that abortSignal, and if a use hits cancel in the middle of operations, the requestDelete should cancel

Reply:
See: https://github.com/ansible/ansible-ui/pull/1169